### DISCUSSION TITLE
Wallet Connect v2 deep link issue with url

### DISCUSSION THREAD

**Discussion by RAcbd** [WalletConnect/walletconnect-monorepo#2947] [Developer Support] [answered]
Supposedly Wallet Connect should be using:

`examplewallet://wc?uri=wc:94caa59c77dae0dd234b5818fb7292540d017b27d41f7f387ee75b22b9738c94@2?relay-protocol=irn&symKey=ce3a2c7724c03cf1769ba8b1bdedad5414cc7b920aa3fb72112b997d1916266f`

It's currently throwing this URL instead:

`examplewallet://wc?uri=wc%94caa59c77dae0dd234b5818fb7292540d017b27d41f7f387ee75b22b9738c94@2?relay-protocol=irn&symKey=ce3a2c7724c03cf1769ba8b1bdedad5414cc7b920aa3fb72112b997d1916266f`

notice the semicolon after "wc" is throwing "%" instead so the deep links are not working at the moment.

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/22793295/9a9e1df7-bef8-4126-8a7d-8c166cf723e9)

You then get this return as it cannot read the current URL.

Is this a change that we need to address on our end?

**Comment by xzilja** [comment]
Deeplinks on android have different format, they use just `wc://uri?=` which prompts andoird's native modal to appear with choices of apps that can handle WalletConnect.

Concrete links like `pitaka://wc?=` are set on ios only.

That said, if you are a user and somehow got to where web3modal generated link like `pitaka://wc?=` on android, that might be a bug, in which case could you please open and issue with further details.

**Reply by RAcbd** [reply]
This was tested on an iOS device, I tried it on Android as well and got the same result. Would you like a sample of deep links being sent through?

**Reply by xzilja** [reply]
Ones provided above are correct, the `%` one you mentioned is in fact preferred format (uri is encoded)

**Reply by RAcbd** [reply]
`pitaka://wc?uri=wc%3A2844d7b751ab6fa7d15b6ba9874d0c8ee3ae7aba8159a00f722c57b7062dd497%402%3Frelay-protocol%3Dirn%26symKey%3D2e7261bb38fbda1f1e77d615e786a5e63683f1629f2dfefb78acd3bcbd118b9d`

This is what's being sent through currently when Pitaka is chosen as a wallet. Previous to this, the deeplinks for Pitaka were working correctly.

I noticed that the URL is "pitaka://wc?uri=wc%" instead of "pitaka://wc?uri=wc:" the latter which should work. Scanning the QR connects properly though.

**Reply by xzilja** [reply]
I'm afraid this is an issue on pitaka wallet side not handling ecnoded uri's. I'll see if our team has contact with to raise this issue, but if you do, please let them know as well.

**Reply by RAcbd** [reply]
Is this a new change though? It was working previously without any updates on Pitaka's end.

**Reply by xzilja** [reply, marked as answer]
Not really, we've been encoding uri's for almost a year now. I think the dapp you were using had very old version of modal before upgraded recently to support our v2 protocol (upgrade led to uri encoiding working).

**Comment by RAcbd** [comment]
> Not really, we've been encoding uri's for almost a year now. I think the dapp you were using had very old version of modal before upgraded recently to support our v2 protocol (upgrade led to uri encoiding working).


This is the correct answer. The DApp I tested on is using an old version of the modal (Opensea, Uniswap). I've tested on DApps that have fully migrated to the newest version of the modal and have confirmed the deeplink works.

Much appreciated @0xAsimetriq

**Comment by kurt-2018** [comment]
Excuse me, how to obtain the parameters behind wc, why the key and public id I applied for are very different from yours

**Comment by Alkhatri77** [comment]
Yeah

**Reply by Alkhatri77** [reply]
Help my friend
